mitiq/pt/pt.py

```diff
@@ -3,28 +3,64 @@
 # This source code is licensed under the GPL license (v3) found in the
 # LICENSE file in the root directory of this source tree.
 
-from typing import Callable, List, Optional, Tuple, Union
+from typing import Callable, List, Optional, Tuple, Union, cast
 
+import random
 import cirq
 import numpy as np
 
 from mitiq import QPROGRAM, Executor, Observable, QuantumResult
 from mitiq.interface import atomic_converter
 
+Paulis = Union[cirq.IdentityGate, cirq.XPowGate, cirq.YPowGate, cirq.ZPowGate]
+
+
+def _generate_lookup_table(
+    gate: str,
+) -> List[Tuple[cirq.Gate, cirq.Gate, cirq.Gate, cirq.Gate]]:
+    if gate not in ["CNOT", "CZ"]:
+        raise ValueError("Invalid two-qubit gate. Supported gates: CNOT, CZ")
+
+    c = cirq.X if gate == "CNOT" else cirq.Z
+    lookup_table = [
+        (cirq.I, cirq.I, cirq.I, cirq.I),
+        (cirq.X, cirq.I, cirq.I, cirq.X),
+        (cirq.Y, cirq.I, cirq.I, cirq.Y),
+        (cirq.Z, cirq.I, cirq.I, cirq.Z),
+        (cirq.I, c, c, cirq.I),
+        (cirq.X, c, c, cirq.X),
+        (cirq.Y, c, c, cirq.Y),
+        (cirq.Z, c, c, cirq.Z),
+        (cirq.I, cirq.Y, cirq.Y, cirq.I),
+        (cirq.X, cirq.Y, cirq.Y, cirq.X),
+        (cirq.Y, cirq.Y, cirq.Y, cirq.Y),
+        (cirq.Z, cirq.Y, cirq.Y, cirq.Z),
+        (cirq.I, cirq.Z, cirq.Z, cirq.I),
+        (cirq.X, cirq.Z, cirq.Z, cirq.X),
+        (cirq.Y, cirq.Z, cirq.Z, cirq.Y),
+        (cirq.Z, cirq.Z, cirq.Z, cirq.Z),
+    ]
+
+    return lookup_table
+
+
+lookup_table_CNOT = _generate_lookup_table("CNOT")
+lookup_table_CZ = _generate_lookup_table("CZ")
+
 
 def execute_with_pauli_twirling(
     circuit: QPROGRAM,
     executor: Union[Executor, Callable[[QPROGRAM], QuantumResult]],
     observable: Optional[Observable] = None,
     *,
-    num_circuits: int = 100,
+    num_circuits: int = 10,
 ) -> float:
     executor = (
         executor if isinstance(executor, Executor) else Executor(executor)
     )
     twirled_circuits = pauli_twirled_circuits(circuit, num_circuits)
     expvals = executor.evaluate(twirled_circuits, observable)
-    return np.average(expvals)
+    return cast(float, np.average(expvals))
```

Comment:
I saw this mypy error as well, but I don't quite understand it.
```
mitiq/pt/pt.py:27: error: Incompatible return value type (got "floating[Any]", expected "float")  [return-value]
```
Do you know what can be classified as `floating[Any]` that's not a `float`?

Totally okay with having a `cast` call here, but I like to understand if it's needed.

Reply:
I wasn't sure why; I just saw the same cast on `np.average()` in `scale_factor_to_expectation_value()` and `execute_with_pec()` and matched the pattern